packages/jazz-tools/src/tools/coValues/request.ts

```diff
@@ -0,0 +1,617 @@
+   */
+  url: string;
+
+  /**
+   * The id of the worker Account or Group.
+   */
+  workerId: string;
+
+  /**
+   * Schema definition for the request payload.
+   * Can be either a direct schema object or an object with schema and optional resolve properties.
+   * The schema defines the structure and validation rules for incoming requests.
+   */
+  request: RequestSchemaDefinition<
+    RequestShape,
+    ResolveQueryStrict<CoMapSchema<RequestShape>, RequestResolve>
+  >;
+
+  /**
+   * Schema definition for the response payload.
+   * Can be either a direct schema object or an object with schema and optional resolve properties.
+   * The schema defines the structure and validation rules for outgoing responses.
+   */
+  response: RequestSchemaDefinition<
+    ResponseShape,
+    ResolveQueryStrict<CoMapSchema<ResponseShape>, ResponseResolve>
+  >;
+}
+
+type AsNullablePayload<T extends MessageShape> = T extends Record<string, never>
+  ? undefined
+  : never;
+type MessageValuePayload<T extends MessageShape> =
+  | Simplify<CoMapInitZod<T>>
+  | AsNullablePayload<T>;
+
+function createMessageEnvelope<S extends MessageShape>(
+  schema: AnyCoMapSchema,
+  value: MessageValuePayload<S>,
+  owner: Account,
+  sharedWith: Account | Group,
+): Loaded<CoMapSchema<S>> {
+  const group = Group.create({ owner });
+
+  group.addMember(sharedWith, "reader");
+
+  // @ts-expect-error - AnyCoMapSchema doesn't have static methods
+  return schema.create(value ?? {}, group);
+}
+
+/**
+ * Function that exports the input CoValue in a serializable format and prepares the information
+ * required for the other side to safely verify the identity of the sender.
+ */
+async function serializeMessagePayload({
+  schema,
+  resolve,
+  value,
+  owner,
+  target,
+}: {
+  // Skipping type validation here to avoid excessive type complexity that affects the typecheck performance
+  schema: AnyCoMapSchema;
+  resolve: any;
+  value: any;
+  owner: Account;
+  target: Account | Group;
+}) {
+  const me = owner ?? Account.getMe();
+  const node = me._raw.core.node;
+  const crypto = node.crypto;
+
+  const agent = node.getCurrentAgent();
+  const signerID = agent.currentSignerID();
+  const signerSecret = agent.currentSignerSecret();
+
+  const envelope = createMessageEnvelope(schema, value, me, target);
+
+  const contentPieces =
+    (await exportCoValue(schema, envelope.id, {
+      resolve,
+      loadAs: me,
+      bestEffortResolution: true,
+    })) ?? [];
+
+  const createdAt = Date.now();
+
+  const signPayload = crypto.secureHash({
+    contentPieces,
+    id: envelope.id,
+    createdAt,
+    signerID,
+  });
+
+  const authToken = crypto.sign(signerSecret, signPayload);
+
+  return {
+    contentPieces,
+    id: envelope.id,
+    createdAt,
+    authToken,
+    signerID,
+  };
+}
+
+const requestSchema = z.object({
+  contentPieces: z.array(z.json()),
+  id: z.custom<`co_z${string}`>(
+    (value) => typeof value === "string" && value.startsWith("co_z"),
```

Comment:
Nit: I see we're doing this check in several places. Could reuse cojson's `isRawCoID` (if it's exported), or define a similar fn for jazz-tools

Reply:
Good point, created a new utility in jazz-tools!